regidx.c

```diff
@@ -294,23 +467,30 @@ int regidx_parse_bed(const char *line, char **chr_beg, char **chr_end, reg_t *re
 
     char *se = ss;
     while ( *se && !isspace_c(*se) ) se++;
-    if ( !*se ) { hts_log_error("Could not parse bed line: %s", line); return -2; }
 
     *chr_beg = ss;
     *chr_end = se-1;
 
+    if ( !*se )
+    {
+        // just the chromosome name
+        *beg = 0;
+        *end = MAX_COOR_0;
```

Comment:
This line can cause high memory usage, but only in an very esoteric scenario.

If we have a region file which is a mix of whole chromosomes and specific regions in those same chromosomes, then `reglist_build_index_` is called and it allocates around a Gb per chromosome.  I ran it on some mouse genome data with all the chromosomes listed twice and it started swapping the 48Gb memory on my system.

However this is VERY specific.  I expect either we'll be specifying regions which are whole chromosomes and not parts of them - in which case we have just 1 entry in the list and the index isn't built - or we'll be specifying lots of regions in each chromosome in which case the end coordinate is known and it works fine (as before).

It's only the mix of whole chromosomes plus at least one sub-region in that same chromosome which triggers the indexing code.

I'm inclined to just reduce the value of MAX_COOR_0 a bit (say 64Gb instead of 1Tb) and just leave it at that.  If it ever becomes an issue we can deal with it differently later, or if we feel it is a concern, merge anyway and raise an issue so we can track it.

This is not a regression as the current code cannot handle this mix of regions anyway.

Reply:
Size has been reduced, and I've also changed the way the memory gets allocated so it may be a bit more efficient.